### ISSUE TITLE
Default role does not detect default roles starting with a single quote

### ISSUE THREAD

**Issue by JulienPalard** [sphinx-contrib/sphinx-lint#40]
For example in:

    Lines containing `'RESTART'` mean that the user execution process has been\n

The `default_role_re` (currently ``re.compile('(^| )`\\w([^`]*?\\w)?`($| )')``) won't catch it due to the `\W` not matching the quote.

**Comment by ezio-melotti** [comment]
The regex also doesn't detect default roles that are followed by a `)`, `,`, `.`, etc., or preceded by a `(` and similar characters -- it only detects them if they are preceded/followed by a space or if they are at the beginning/end of the line:
```shell
$ echo 'This is not detected: (`foo`)' >> buggy.rst; sphinx-lint --enable default-role buggy.rst
No problems found.
$ echo 'This is detected: ( `foo` )' >> buggy.rst; sphinx-lint --enable default-role buggy.rst
buggy.rst:2: default role used (hint: for inline literals, use double backticks) (default-role)
$ echo 'This is not detected: `foo`, `bar`.' >> buggy.rst; sphinx-lint --enable default-role buggy.rst
buggy.rst:2: default role used (hint: for inline literals, use double backticks) (default-role)
$ echo 'This is detected: `foo` , `bar`.' >> buggy.rst; sphinx-lint --enable default-role buggy.rst
buggy.rst:2: default role used (hint: for inline literals, use double backticks) (default-role)
buggy.rst:4: default role used (hint: for inline literals, use double backticks) (default-role)
$ cat buggy.rst 
This is not detected: (`foo`)
This is detected: ( `foo` )
This is not detected: `foo`, `bar`.
This is detected: `foo` , `bar`.
```

**Comment by ezio-melotti** [comment]
Another [real world example](https://github.com/python/cpython/pull/96445#discussion_r973613734) I met today:
```shell
$ echo 'The only supported backend is `"perf"`.' > buggy.rst
$ sphinx-lint --enable default-role buggy.rst 
No problems found.
$ echo 'The only supported backend is `perf`.' >> buggy.rst
$ sphinx-lint --enable default-role buggy.rst 
No problems found.
```

**Comment by ezio-melotti** [comment]
[Another one](https://github.com/python/cpython/pull/96924/commits/2d6870740429c2118f3fd2c3ef1d233cb1cdc4fe):
```sh
$ echo 'Assignment expressions using the walrus operator `:=` assign a variable' > buggy.rst
$ sphinx-lint --enable default-role buggy.rst 
No problems found.

```
